server/database_manager/src/main.rs

```diff
@@ -2,60 +2,37 @@
 
 // Std.
 use std::env;
-use std::fs;
 use std::io;
 use std::io::*;
 
 // Custom.
 use shared::misc::db_manager::*;
 
-/// Looks if there is a database file in the current directory.
-/// Returns `true` if a database was found, `false` if not.
-fn find_database() -> bool {
-    let paths = fs::read_dir(".");
-    if let Err(e) = paths {
-        panic!("{}", e);
-    }
-    let paths = paths.unwrap();
-
-    for path in paths {
-        if let Err(e) = path {
-            panic!("{}", e);
-        }
-        let path = path.unwrap();
-
-        if path.file_type().unwrap().is_file()
-            && path.file_name().to_str().unwrap() == DATABASE_NAME
-        {
-            return true;
-        }
-    }
-
-    false
-}
-
 fn main() {
     println!(
         "FBugReporter (database manager) (v{}).",
         env!("CARGO_PKG_VERSION")
     );
     println!("Type 'help' to see commands...\n");
 
-    if !find_database() {
+    let database_location = DatabaseManager::get_database_location();
+
+    println!(
+        "Looking for database in '{}'...",
+        database_location.to_string_lossy()
+    );
+
+    if !(database_location.exists() && database_location.is_file()) {
         println!(
-            "No '{}' file was found in the current directory, make sure \
+            "No database file found at '{}', make sure \
             the database is created (the server will create it once it's started \
             for the first time).",
-            DATABASE_NAME
+            database_location.to_string_lossy()
         );
         return;
     }
 
```

Comment:
If the database was found add:
```Rust
    } else {
        println!(
            "INFO: Found database at '{}'",
            database_location.to_string_lossy()
        );
    }
```
otherwise it looks like its still searching and searching for the database.

Reply:
Fair point.